extensions/wikia/BannerNotifications/js/BannerNotifications.js

```diff
@@ -0,0 +1,269 @@
+		 */
+		function setContent(content) {
+			bannerNotification
+				.$element
+				.find('.msg')
+				.html(content);
+		}
+
+		/**
+		 * Changes type of the notification to the provided one
+		 * @param {String} type
+		 */
+		function setType(type) {
+			if (types.hasOwnProperty(type)) {
+				bannerNotification
+					.$element
+					.removeClass(classes)
+					.addClass(type);
+			}
+		}
+
+		bannerNotification = {
+			$element: createMarkup(content, type),
+			hide: hide,
+			setContent: setContent,
+			setType: setType
+		};
+
+		isModal = isModalShown();
+
+		// Modal notifications have no close button so set a timeout
+		if (isModal && typeof timeout !== 'number') {
+			timeout = defaultTimeout;
+		}
+
+		bannerNotification.setType(type);
+
+		addToDOM(
+			bannerNotification.$element,
+			$parent
+		);
+
+		// Share scroll event with WikiaFooterApp's toolbar floating (BugId:33365)
+		if (window.WikiaFooterApp) {
+			window.WikiaFooterApp.addScrollEvent();
+		}
+
+		// Close notification after specified amount of time
+		if (typeof timeout === 'number') {
+			setTimeout(function () {
+				hide();
+			}, timeout);
+		}
+
+		return bannerNotification;
+	}
+
+	/**
+	 * Shows notification informing about an AJAX connection error
+	 * @returns {Object}
+	 */
+	function showConnectionError() {
+		return show(
+			$.msg('bannernotifications-general-ajax-failure'),
+			'error'
+		);
+	}
+
+	/**
+	 * Called once to instantiate this feature
+	 */
+	function init() {
+		var pageContainerSelector =
+			window.skin === 'monobook' ? '#content' : '.WikiaPageContentWrapper';
+		setUpClose();
+
+		pageContainer = $(pageContainerSelector)[0];
+		wikiaHeader = $('#globalNavigation');
+		headerHeight = wikiaHeader.height();
+	}
+
+	/**
+	 * Determine if a modal is present and visible so we can apply the notification to the modal instead of the page.
+	 * @returns {boolean}
+	 */
+	function isModalShown() {
+		// handle all types of modals since the begining of time!
+		modal = $('.modalWrapper, .yui-panel, .modal');
+
+		// returns false if there's no modal or it's hidden
+		return modal.is(':visible');
+	}
+
+	/**
+	 * Removes notification element from DOM and executes callback
+	 * @param {jQuery} $elements - elements to be removed from DOM
+	 * @param {function} [callback]
+	 */
+	function removeFromDOM($elements, callback) {
+		if ($elements.length) {
+			$elements.animate({
+				'height': 0,
+				'padding': 0,
+				'opacity': 0
+			}, 400, function () {
+				$elements.remove();
+				if (typeof callback === 'function') {
+					callback();
+				}
+			});
+		} else {
+			if (typeof callback === 'function') {
+				callback();
+			}
+		}
+	}
+
+	/**
+	 * Hides all displayed notifications
+	 * @param {Function} callback
+	 */
+	function hideAll(callback) {
+		removeFromDOM($('.banner-notification'), callback);
+	}
+
+	/**
+	 * Bind close event to close button
+	 */
+	function setUpClose() {
+		$(document.body).on('click', '.close', onCloseClicked);
```

Comment:
The selector `.close` is probably not specific enough, since there's going to be `.close` elements in places other than banner notifications, and there may be unexpected consequences if `onCloseClicked` is fired on elements like modals.